### PR TITLE
fix: Handle external type changes in stream resources

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- handle external type changes in stream resources

## Test Plan

* [x] acceptance tests

* [ ] …

### CODE REVIEW

File: pkg/resources/custom_diffs.go [hunk 1]
```diff
@@ -228,6 +228,35 @@ func RecreateWhenSecretTypeChangedExternally(secretType sdk.SecretType) schema.C
...
+			// if the type is empty, the state is empty - do not recreate
+			if gotTypeRaw == "" {
+				return nil
+			}
+
+			gotType, err := toType(gotTypeRaw)
+			if err != nil {
+				return fmt.Errorf("unknown type: %w", err)
+			}
+			if gotType != wantType {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/custom_diffs.go:252]: please add the same comment as in the user specific func here.

nit: can we reuse this func in user and in secret (secret has some additional logic; user empty type is a valid case AFAIK so differs from this impl - but still, for me it makes sense if we can reuse the function and currently was not reused in other places)?

File: pkg/resources/stream_on_directory_table_acceptance_test.go [hunk 2]
```diff
@@ -459,3 +459,56 @@ func TestAcc_StreamOnDirectoryTable_InvalidConfiguration(t *testing.T) {
...
+				),
+			},
+			// external change with a different type
+			{
+				PreConfig: func() {
+					table, cleanupTable := acc.TestClient().Table.CreateWithChangeTracking(t)
+					t.Cleanup(cleanupTable)
+					acc.TestClient().Stream.DropFunc(t, id)()
+					_, cleanup := acc.TestClient().Stream.CreateOnTableWithRequest(t, sdk.NewCreateOnTableStreamRequest(id, table.ID()))
+					t.Cleanup(cleanup)
```
Comment (sfc-gh-asawicki) [inline pkg/resources/stream_on_directory_table_acceptance_test.go:495]: nit: let's assert in this preconfig that after it we have other type

**Review by sfc-gh-asawicki — DISMISSED** [review]

File: pkg/resources/custom_diffs.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/resources/custom_diffs.go:252]: Added a comment.

Yeah, initially, I thought about reusing this even more, but I decided not to spend more time on it - I can do it in the next PR.

File: pkg/resources/stream_on_directory_table_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/resources/stream_on_directory_table_acceptance_test.go:496]: Done.

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**